File: 3945-digit-frequency-score/3945-digit-frequency-score.py
from collections import defaultdict
class Solution:
    def digitFrequencyScore(self, n: int) -> int:
        freq = defaultdict(int)
        while(n > 0):
            dig = n % 10
            freq[dig] += 1
            n //= 10
        s = 0
        for k in freq:
            s += (k * freq[k])
        return int(s)


        

File: 3945-digit-frequency-score/3945-digit-frequency-score.cpp

class Solution {
public:
    int digitFrequencyScore(int n) {
        map<int,int>mp; int s = 0;
       while(n > 0){
        int d = n % 10;
            mp[d]++;
            n /= 10;
       }
        for(auto [k,v] : mp){
            s += (k * v);
        }

        return s;
    }
};